integration/spark/src/test/scala/org/apache/carbondata/spark/testsuite/addsegment/AddSegmentTestCase.scala

```diff
@@ -781,6 +781,26 @@ class AddSegmentTestCase extends QueryTest with BeforeAndAfterAll {
     sql(s"drop table $tableName")
   }
 
+  test("Test add segment by carbon written by sdk having old timestamp") {
+    sql(s"drop table if exists external_primitive")
+    sql(
+      s"""
+         |create table external_primitive (id int, name string, rank smallint, salary double,
+         | active boolean, dob date, doj timestamp, city string, dept string) stored as carbondata
+         |""".stripMargin)
+    val externalSegmentPathWithOldTimestamp = storeLocation + "/" +
+      "external_segment_with_old_timestamp"
+    val externalSegmentPath = storeLocation + "/" + "external_segment"
+    FileFactory.deleteAllFilesOfDir(new File(externalSegmentPath))
+    copy(externalSegmentPathWithOldTimestamp, externalSegmentPath)
```

Comment:
We should not keep binary files in the repo (like carbondata and index files). It has to generate on the fly.
Try to write from SDK and use nano time timeStamp by calling `CarbonWriterBuilder#uniqueIdentifier`

Reply:
Done